service/controller/clusterapi/v28/machine_deployment_resource_set.go

```diff
@@ -0,0 +1,132 @@
+package v28
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/controller/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/controller/resource/retryresource"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/credential"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/key"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/resource/asgstatus"
+)
+
+type MachineDeploymentResourceSetConfig struct {
+	ControlPlaneAWSClients aws.Clients
+	G8sClient              versioned.Interface
+	K8sClient              kubernetes.Interface
+	Logger                 micrologger.Logger
+
+	HostAWSConfig  aws.Config
+	ProjectName    string
+	Route53Enabled bool
+}
+
+func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig) (*controller.ResourceSet, error) {
+	var err error
+
+	var asgStatusResource controller.Resource
+	{
+		c := asgstatus.Config{
+			G8sClient: config.G8sClient,
+			Logger:    config.Logger,
+		}
+
+		asgStatusResource, err = asgstatus.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []controller.Resource{
+		asgStatusResource,
```

Comment:
That is only to have some resource since the `Resources` argument must not be empty. We will change this later but for now it should be good enough.